src/shared/meson.build

```diff
@@ -71,6 +71,8 @@ shared_sources = files(
         'clean-ipc.h',
         'clock-util.c',
         'clock-util.h',
+        'compare-operator.c',
```

Comment:
this new module could use a new unit test? just to encode expectations, which are in pid1 Condition and so on, to ensure we don't break anything accidentally

Reply:
the existing condition tests thoroughly test the compare stuff, I think we have roughly 1000% coverage already on this, not sure it's worth adding even more for that ;-)